index.js

```diff
@@ -237,11 +237,19 @@ RedisClient.prototype.create_stream = function () {
             this.stream.destroy();
         }
 
+        // Accept snake_case options and convert them to camelCase
+        var opts = {};
+        var keys = Object.keys(this.connection_options);
+        var key;
+        while (key = keys.pop()) {
+            opts[key.replace(/_([a-z])/g, function (g) { return g[1].toUpperCase(); })] = this.connection_options[key];
+        }
```

Comment:
This is going to run on every reconnect attempt. Instead, the fix could be placed where the options are converted to camel_case. That would only run once for each client. The clone function could receive a extra argument to skip the converting tls options to camel_case and special handle that by checking if the key name is `tls`.